### PR TITLE
feat: adding support for api event request

### PR DESCRIPTION
**Pull request by inavac182**
## Description

When using AWS_PROXY, the request from the api will include everything in the request so the lambda need to adapt to support the new response. This PR is to update the lambda to work with the api request.

I'm adding a few helpers to help parse the stage variables and also mocks for unit tests.

### CODE REVIEW

File: functions/__mocks__/mocked-event.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { APIGatewayEvent } from "aws-lambda";
+import { StageEnvironment } from "../common";
+
+export const getMockedEvent = (body: string, environment: StageEnvironment = 'test'): APIGatewayEvent => {
+  return {
```
[the remark this answers is not in the document]
Comment (saudelog) [reply functions/__mocks__/mocked-event.ts:5]: nice!, I'm now thinking that we could directly use stage from the requestContext to select the table, that way we avoid relaying on manually having to set stageVariables, that will also give use a chance to have more stages with other tables

[the remark this answers is not in the document]
Comment (inavac182) [reply functions/__mocks__/mocked-event.ts:5]: Yeah, although I wasn't able to see the stage param in the real request:

Screenshot 2023-06-29 at 12 21 43 PM

Idk if there is a config to get that in the request context?

[the remark this answers is not in the document]
Comment (inavac182) [reply functions/__mocks__/mocked-event.ts:5]: That is truncated so we might need to console log that request to verify what we are getting there

[the remark this answers is not in the document]
Comment (saudelog) [reply functions/__mocks__/mocked-event.ts:5]: Would it work if we use stage:'test-invoke-stage' ? Wondering how that value looks like once we use a sub domain on each stage

[the remark this answers is not in the document]
Comment (saudelog) [reply functions/__mocks__/mocked-event.ts:5]: I tried with the the Invoke URL of 2 different stages `test` and `testv2`, I got those as values on the stage in the request context, it seems like the value is different when invoking using the test section of the api


**stage: test**
Screenshot 2023-06-29 at 1 22 04 PM

**stage: testv2**
Screenshot 2023-06-29 at 1 24 29 PM

**Review by saudelog — APPROVED** [review]

**Pull request merged**